book/listings/todo_app/1/todo_object/mod.rs

```diff
@@ -0,0 +1,36 @@
+mod imp;
+
+use glib::Object;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use serde::{Deserialize, Serialize};
+
+glib::wrapper! {
+    pub struct TodoObject(ObjectSubclass<imp::TodoObject>);
+}
+
+impl TodoObject {
+    pub fn new(content: String, completed: bool) -> Self {
+        Object::new(&[("content", &content), ("completed", &completed)])
+            .expect("Failed to create `TodoObject`.")
+    }
+
+    pub fn is_completed(&self) -> bool {
+        self.property("completed")
+            .expect("The property needs to exist and be readable.")
+            .get()
+            .expect("The property needs to be of type `bool`.")
+    }
+
+    pub fn todo_data(&self) -> TodoData {
+        let imp = imp::TodoObject::from_instance(self);
+        imp.data.borrow().clone()
```

Comment:
Can't you just pass a reference here? do you need TodoData to be Clone?

Reply:
Yep, otherwise the borrow checker yells at me. You are very welcome to play with it though 😁

Reply:
Then I would wrap it in a Rc

Reply:
Done